main.go

```diff
@@ -31,19 +31,26 @@ func init() {
 	}
 }
 
-func main() {
-	err := app.Flag.Parse(os.Args[1:])
+func runArgs(argsStr []string) error {
+	err := app.Flag.Parse(argsStr)
 	if err != nil {
 		log.Printf("error parsing flags: %v\n", err)
-		os.Exit(1)
+		return err
 	}
 
 	args := app.Flag.Args()
 	err = app.Dispatch(args)
 	if err != nil {
 		log.Printf("error dispatching command: %v\n", err)
-		os.Exit(1)
+		return err
 	}
+	return nil
+}
 
+func main() {
+	err := runArgs(os.Args[1:])
+	if err != nil {
+		os.Exit(1)
```

Comment:
and now, I would call:

```go
log.Print(err)
os.Exit(1)
```

or, directly:
```go
log.Fatal(err)
```

Reply:
Done.